3648/1/gem_stat.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'open-uri'
+require 'httparty'
+
+AUTH_TOKEN = YAML.load_file('tokens.yml')['AUTH_TOKEN']
+
+class GemStatistic
+  attr_reader :gem_name
+  def initialize(gem_name)
+    @gem_name = gem_name
+  end
+
+  def create_stat
+    {
+      gem_name: gem_name,
+      used_by: find_used_by,
+      watched_by: find_watched_by,
+      stars: find_stars,
+      forks: find_forks,
+      contributors: find_contributors,
+      open_issues: find_issues
+    }
+  end
+
+  private
+
+  def find_repo
+    url = "https://#{AUTH_TOKEN}@api.github.com/search/repositories?q=#{gem_name}&per_page=1"
```

Comment:
Move logic for API call to separate class (and file). It is better when class does only one particular job. If API changes in future, you will find that logic easily and will need to change only this class